pubtools/_content_gateway/push_staged.py

```diff
@@ -1,2 +1,191 @@
+                    return
+                LOG.info(f"Product record found with product_id:- {product_id}")
+                item.get('metadata')['id'] = product_id
+                LOG.info("Updating product metadata")
+                self.cgw_client.update_product(item.get('metadata'))
+            elif item.get('state') == 'absent' and product_id:
+                LOG.info(f"Deleting existing product records for product_id:- {product_id}")
+                self.cgw_client.delete_product(product_id)
+                LOG.info("Product record deleted!")
+            else:
+                LOG.warning("Invalid state or metadata!")
+        except Exception as err:
+            LOG.exception(err)
+
+    def process_version(self, item):
+        try:
+            product_name = item.get('metadata')['productName']
+            product_code = item.get('metadata')['productCode']
+            version_name = item.get('metadata')['versionName']
+            product_id = self.__get_product_id(product_name, product_code)
+            LOG.debug(f"Fetching the product and version id of "
+                      f"product name- {product_name} and version name- {version_name}")
+            version_id = self.__get_version_id(product_name, product_code, version_name)
+
+            if item.get('state') == 'present':
+                item.get('metadata')['productId'] = product_id
+                del item.get('metadata')['productName'], item.get('metadata')['productCode']
+                if not version_id:
+                    LOG.info("No previous entries found for the given product's version")
+                    LOG.info("Creating version entry for the given version metadata")
+                    version_id = self.cgw_client.create_version(product_id, item.get('metadata'))
+                    self.__create_product_version_mapping(product_name, product_code)
+                    LOG.info(f"New version created with version_id:- {version_id}")
+                    return
+                item.get('metadata')['id'] = version_id
+                LOG.info(f"Found version record with version_id:- {version_id}")
+                LOG.info("Updating the version metadata")
+                self.cgw_client.update_version(product_id, item.get('metadata'))
+            elif item.get('state') == 'absent' and version_id:
+                LOG.info(f"Deleting existing version records for version_id:- {version_id}")
+                self.cgw_client.delete_version(product_id, version_id)
+                LOG.info("Version record deleted!")
+            else:
+                LOG.warning("Invalid state or metadata!")
+        except Exception as error:
+            LOG.exception(error)
+
+    def process_file(self, file_item):
+        try:
+            product_id, version_id, file_id = self.__get_file_id(file_item)
+            LOG.debug(f"Fetching file id for the file metadata")
+            if not (product_id and version_id):
```

Comment:
If the user is trying to create a new file, in that case, it will return None for all the ids (i.e product, version and file id).
The mapping is like this {(product_name, product_code, version_name, downloadURL): (product_id, version_id, file_id)}. This __get_file_id() will return the ids only for the file update and delete operation.